### ISSUE TITLE
Unstable home detection

### ISSUE THREAD

**Issue by wigster** [mudape/iphonedetect#66]
Hi,

We have two iphone 7s, both tracked by iphonedetect. Mine  works exactly as it should: switches to off a few minutes after I leave, comes back immediately when I reconnect. My wife's during most of the day works fine, but does this during clustered periods in the day when it is just lying there without being moved:
```
Was detected away
11:46:55 - 16 seconds ago
Was detected at home
11:42:29 - 5 minutes ago
Was detected away
11:41:05 - 6 minutes ago
Was detected at home
11:34:15 - 13 minutes ago
Was detected away
11:34:05 - 13 minutes ago
Was detected at home
11:30:26 - 17 minutes ago
Was detected away
11:30:05 - 17 minutes ago
Was detected at home
11:28:25 - 19 minutes ago
Was detected away
11:28:05 - 19 minutes ago
```
and then it is stable for the rest of the 24-hour period. But I cannot for the life of me think of what could be causing it.

Both the phones get static IPs on the home network from DHCP. The config looks like this:

```
device_tracker:
  - platform: iphonedetect
    consider_home: 60
    scan_interval: 12
    new_device_defaults:
      track_new_devices: true
    hosts:
      idevice_wifesphone: 192.168.0.8
      idevice_myphone: 192.168.0.9
```
Would you have any suggestions what could be causing this?

**Comment by mastakebob** [comment]
Just chiming in to say that I'm also seeing the same behavior with my wife's iphone SE.  Seems to have emerged in the last few months (the previous year+ of use was rock solid accurate).  Getting the same home/away jitter when the iphone SE is in the house on the network.  Didn't do any jitter from 8pm to 8am last night, but this morning it's been going away/coming home every 30 minutes or so for the last 4 hours.

**Comment by kukasme** [comment]
I have the same problem. 3x Iphones (2xXS and 1xSE) and they all disconnect and reconnect.

**Comment by toddaustin07** [comment]
I was excited to have found this technique, but unfortunately it just isn't reliable.  Maybe it worked on earlier iphone models and older iOS versions, but with my iPhone 12 and iOS 15 results are too inconsistent.  When home and running on battery, after a few hours (presumably it goes into deep sleep), it can appear offline for a couple minutes.  And when returning home with the phone asleep, it's hit and miss whether it gets picked back up again until you manually wake the phone up.  I don't think sending UDP messages to port 5353 is an effective method to force ARP table updates.

**Comment by mudape** [comment]
In our home I've noticed:
An iPhone 7 was loosing connection when roaming between access points. This Christmas that phone was replaced and the new one, which moves in the same area, is reporting correct state.
An iPhone 12 is loosing connection a couple of minutes from time to time, but only when it is (or should be..) connected to the main access point.

So, I wouldn't say this "trick" is totally broken with recent iOS updates but yes it's more unstable.
That seems to be on the device/network though and hence hard to fix with this integration.
